components/ai_chat/browser/engine/engine_consumer_claude.cc

```diff
@@ -0,0 +1,207 @@
+  // provided here.
+  const auto model_name = ai_chat::features::kAIModelName.Get();
+  base::flat_set<base::StringPiece> stop_sequences(kStopSequences.begin(),
+                                                   kStopSequences.end());
+  api_ = std::make_unique<RemoteCompletionClient>(model_name, stop_sequences,
+                                                  url_loader_factory);
+}
+
+EngineConsumerClaudeRemote::~EngineConsumerClaudeRemote() = default;
+
+void EngineConsumerClaudeRemote::ClearAllQueries() {
+  api_->ClearAllQueries();
+}
+
+void EngineConsumerClaudeRemote::GenerateQuestionSuggestions(
+    const bool& is_video,
+    const std::string& page_content,
+    SuggestedQuestionsCallback callback) {
+  std::string prompt;
+  std::vector<std::string> stop_sequences;
+  prompt = base::StrCat(
+      {kHumanPromptSequence,
+       base::ReplaceStringPlaceholders(
+           l10n_util::GetStringUTF8(is_video
+                                        ? IDS_AI_CHAT_VIDEO_PROMPT_SEGMENT
+                                        : IDS_AI_CHAT_ARTICLE_PROMPT_SEGMENT),
+           {page_content}, nullptr),
+       "\n\n", l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_PROMPT_SEGMENT),
+       kAIPromptSequence, "<response>"});
+  CheckPrompt(prompt);
+  stop_sequences.push_back("</response>");
+
+  api_->QueryPrompt(
+      prompt, stop_sequences,
+      base::BindOnce(
+          &EngineConsumerClaudeRemote::OnGenerateQuestionSuggestionsResponse,
+          weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void EngineConsumerClaudeRemote::OnGenerateQuestionSuggestionsResponse(
+    SuggestedQuestionsCallback callback,
+    GenerationResult result) {
+  if (!result.has_value() || result->empty()) {
+    // Query resulted in error
+    LOG(ERROR) << "Error getting question suggestions.";
+    return;
+  }
+
+  // Success
+  std::vector<std::string> questions =
+      base::SplitString(*result, "|", base::WhitespaceHandling::TRIM_WHITESPACE,
+                        base::SplitResult::SPLIT_WANT_NONEMPTY);
+  std::move(callback).Run(std::move(questions));
+}
+
+void EngineConsumerClaudeRemote::GenerateAssistantResponse(
+    const bool& is_video,
+    const std::string& page_content,
+    const ConversationHistory& conversation_history,
+    const std::string& human_input,
+    GenerationDataCallback data_received_callback,
+    GenerationCompletedCallback completed_callback) {
+  std::string prompt = BuildClaudePrompt(human_input, page_content, is_video,
+                                         conversation_history);
+  CheckPrompt(prompt);
+  api_->QueryPrompt(prompt, {"</response>"}, std::move(completed_callback),
+                    std::move(data_received_callback));
+}
+
+void EngineConsumerClaudeRemote::SanitizeInput(std::string& input) {
+  base::ReplaceSubstringsAfterOffset(&input, 0, kHumanPrompt, "");
+  base::ReplaceSubstringsAfterOffset(&input, 0, kAIPrompt, "");
```

Comment:
@LorenzoMinto do you think this is ok that we're stripping `Human:` from any input (article or human entered) and not waiting for the full `\n\nHuman: `? I think the reason to do it is in case something does have the smaller string, without the perfect line breaks and Claude still interprets it as a change of character. Downside is that it's over-eager and could produce unexpected results when we're silently ignoring a substring.